tests/unit/cag/test_inequality.py

```diff
@@ -0,0 +1,973 @@
+        with pytest.raises(PatternNotMetError, match=err_msg):
+            instance._validate_pattern_with_data(data, metadata)
+
+    def test__get_updated_metadata(self):
+        """Test the ``_get_updated_metadata`` method."""
+        # Setup
+        instance = Inequality(low_column_name='low', high_column_name='high')
+        metadata = Metadata.load_from_dict({
+            'tables': {
+                'table': {
+                    'columns': {
+                        'low': {'sdtype': 'numerical'},
+                        'high': {'sdtype': 'numerical'},
+                        'col': {'sdtype': 'id'},
+                    },
+                    'column_relationships': [
+                        {'type': 'relationship', 'column_names': ['low', 'high']},
+                        {'type': 'relationship', 'column_names': ['low', 'col']},
+                        {'type': 'relationship', 'column_names': ['high', 'col']},
+                    ],
+                }
+            }
+        })
+
+        # Run
+        updated_metadata = instance._get_updated_metadata(metadata)
+
+        # Assert
+        expected_metadata_dict = Metadata.load_from_dict({
+            'tables': {
+                'table': {
+                    'columns': {
+                        'low': {'sdtype': 'numerical'},
+                        'col': {'sdtype': 'id'},
+                        'low#high': {'sdtype': 'numerical'},
+                    },
+                    'column_relationships': [
+                        {'type': 'relationship', 'column_names': ['low', 'col']},
+                    ],
+                }
+            }
+        }).to_dict()
+        assert updated_metadata.to_dict() == expected_metadata_dict
+
+    def test__fit(self):
+        """Test it learns the correct attributes."""
+        # Setup
+        table_data = {'table': pd.DataFrame({'a': [1, 2, 4], 'b': [4, 5, 6]})}
+        metadata = Metadata.load_from_dict({
+            'tables': {
+                'table': {
+                    'columns': {
+                        'a': {'sdtype': 'datetime', 'datetime_format': '%y %m, %d'},
+                        'b': {'sdtype': 'datetime', 'datetime_format': '%y %m %d'},
+                    },
+                }
+            }
+        })
+        instance = Inequality(low_column_name='a', high_column_name='b', table_name='table')
+
+        # Run
+        instance._fit(table_data, metadata)
+
+        # Assert
+        assert instance._is_datetime is True
+        assert instance._dtype == pd.Series([1]).dtype  # exact dtype (32 or 64) depends on OS
+        assert instance._low_datetime_format == '%y %m, %d'
+        assert instance._high_datetime_format == '%y %m %d'
+
+    def test__fit_numerical(self):
+        """Test it for numerical columns."""
+        # Setup
+        table_data = {'table': pd.DataFrame({'a': [1, 2, 4], 'b': [4.0, 5.0, 6.0]})}
```

Comment:
Could we parameterize the input here and test that it works with follow pandas dtypes?
- Float64
- Float32
- Int64
- Int32
- Int16
- Int8